modules/wolfgame.py

```diff
@@ -787,6 +798,249 @@ def fpinger(cli, nick, chan, rest):
+                else:
+                    msg.append("You will now get pinged once when your preferred amount of players is reached.")
+                    var.PING_PREFS_ACCS[acc] = "once"
+                    var.set_ping_pref_acc(acc, "once")
+            elif cloak in var.PING_PREFS.keys() and var.PING_PREFS[cloak] == "once":
+                msg.append("You are already set to be pinged once when your desired player count is reached.")
+            else:
+                msg.append("You will now get pinged once when your preferred amount of players is reached.")
+                var.PING_PREFS[cloak] = "once"
+                var.set_ping_pref(cloak, "once")
+
+        elif pref.lower() in ("ondemand", "ping", botconfig.CMD_CHAR + "ping"):
+            if acc and acc != "*":
+                if acc in var.PING_PREFS_ACCS.keys() and var.PING_PREFS_ACCS[acc] == "ping":
+                    msg.append("You are already set to be added to the {0}ping list when enough players have joined.")
+                else:
+                    msg.append("You will now be added to the {0}ping list when enough players have joined.")
+                    var.PING_PREFS_ACCS[acc] = "ping"
+                    var.set_ping_pref_acc(acc, "ping")
+            elif cloak in var.PING_PREFS.keys() and var.PING_PREFS[cloak] == "ping":
+                msg.append("You are already set to be added to the {0}ping list when enough players have joined.")
+            else:
+                msg.append("You will now be added to the {0}ping list when enough players have joined.")
+                var.PING_PREFS[cloak] = "ping"
+                var.set_ping_pref(cloak, "ping")
+
+        elif pref.lower() in ("all", "always"):
+            if acc and acc != "*":
+                if acc in var.PING_PREFS_ACCS.keys() and var.PING_PREFS_ACCS[acc] == "all":
+                    msg.append("You are already set to be added to the {0}ping list as well as being pinged immediately when enough players have joined.")
+                else:
+                    msg.append("You will now be added to the {0}ping list as well as being pinged immediately when your preferred amount of players is reached.")
+                    var.PING_PREFS_ACCS[acc] = "all"
+                    var.set_ping_pref_acc(acc, "all")
+            elif cloak in var.PING_PREFS.keys() and var.PING_PREFS[cloak] == "all":
+                msg.append("You are already set to be added to the {0}ping list as well as being pinged immediately when enough players have joined.")
+            else:
+                msg.append("You will now be added to the {0}ping list as well as being pinged immediately when your preferred amount of players is reached.")
+                var.PING_PREFS[cloak] = "all"
+                var.set_ping_pref(cloak, "all")
+        else:
+            msg.append("Invalid parameter. Please enter a non-negative integer or a valid preference.")
```

Comment:
Maybe silently fail here? That way I can do something like "!pingif 6 because I hate small games" -- a lot of our other commands allow that kind of construct where the user can insert arbitrary text after the parameters as kind of an explanatory note; I see no reason to prevent it here as well (especially since the command actually succeeds at setting your pref to 6 in addition to giving you the error message)